### PR TITLE
Added Api

### PR DESCRIPTION
**Pull request by TuyishimireEric**
# Bookstore: connect to API
In this branch, I have connected the React-Redux application to an existing API to create and remove books on a remote server.
>In the list below, I have highlighted all the activities made in this branch.

## Highlights :+1: 
- [x] I have fetched the book list from Bookstore API on Books page load.
- [x] I have added an Action Creator, that returns a function.
- [x] I fetched the list of books.
- [x] I dispatched another action that will pass the API response to the reducer and update the state.
- [x] I refactored the add book and remove book features to persist changes in the server.
- [x] I changed the existing Action Creators, so they return functions.
- [x] I made all changes by using API.

**Cheers and happy reviewing! :v:**

### CODE REVIEW

**Review by DammyShittu — COMMENTED** [review]
Hi @TuyishimireEric,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a project that did not follow the right git flow. You are required to merge the feature branch to the dev branch, not the main

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by TuyishimireEric** [comment]
Thank you @DammyShittu
I have fixed the issue now

File: src/components/Book.js [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
 import { useDispatch } from 'react-redux';
 import { removeBook } from '../redux/books/books';
 
 function Book({ title, author, id }) {
   const dispatch = useDispatch();
+  const apiUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/8bHEqSbfdyYUuS4cM1PJ/books/';
+  async function remove(id) {
+    dispatch(removeBook(id));
+    await axios.delete(`${apiUrl}${id}`);
+  }
```
Comment (vechicin) [inline src/components/Book.js]: This should be an action creator within your Redux `books.js` file. Please fix it.

Screen Shot 2022-08-25 at 7 43 34

**Review by vechicin — CHANGES_REQUESTED** [review]
# Hi @TuyishimireEric ✋, 

Good job so far! 🔥 
There are some issues that you still need to work on to go to the next project but you are almost there! 🚨 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights ⚡ 

- [x] Good job refactoring some of the code!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 📂 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by vechicin** [comment]
https://github.com/TuyishimireEric/bookstore/blob/2befb0f2233e8cad339daced1ac42144adb1304f/src/redux/configureStore.js#L1-L12

According to the project requirements, you should use `createAsyncThunk` for creating thunk action creators.

Screen Shot 2022-08-25 at 7 46 25

You don't have to use the code exactly as written in the example, just make sure you employ the `createAsyncThunk` method. Please make sure you refactor all of your action creators using this method.

File: src/components/Book.js (hunk 1 above)
Comment (TuyishimireEric) [reply src/components/Book.js]: Thank you @vechicin for reviewing my project.

**Review by uwadonat — APPROVED** [review]
## Hi @TuyishimireEric, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
unnamed

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by TuyishimireEric** [comment]
Thank you @uwadonat for approving my project :+1:

**Pull request merged**